include/Interface/buddy/dip/dip.h

```diff
@@ -161,5 +247,197 @@ MemRef<float, 2> Resize2D(Img<float, 2> *input, INTERPOLATION_TYPE type,
+
+  intptr_t sizesOutput[2] = {outputRows, outputCols};
+  MemRef<float, 2> output1(sizesOutput);
+  float *data = new float[outputRows * outputCols];
+  for (int i = 0; i < outputRows * outputCols; ++i) {
+    data[i] = -1;
+  }
+  MemRef<float, 2> copymemref(data, sizesOutput, 0);
+  for (int i = 0; i < outputRows * outputCols; ++i) {
+    data[i] = 256;
+  }
+  MemRef<float, 2> copymemref1(data, sizesOutput, 0);
+  if (option == BOUNDARY_OPTION::CONSTANT_PADDING) {
+    detail::_mlir_ciface_closing_2d_constant_padding(
+        input, kernel, output, &output1, &copymemref, &copymemref1, centerX,
+        centerY, iterations, constantValue);
+  } else if (option == BOUNDARY_OPTION::REPLICATE_PADDING) {
+    detail::_mlir_ciface_closing_2d_replicate_padding(
+        input, kernel, output, &output1, &copymemref, &copymemref1, centerX,
+        centerY, iterations, 0);
+  }
+}
+
+void TopHat2D(Img<float, 2> *input, MemRef<float, 2> *kernel,
+              MemRef<float, 2> *output, unsigned int centerX,
+              unsigned int centerY, unsigned int iterations,
+              BOUNDARY_OPTION option, float constantValue = 0) {
+  intptr_t outputRows = output->getSizes()[0];
+  intptr_t outputCols = output->getSizes()[1];
+  intptr_t sizesOutput[2] = {outputRows, outputCols};
+  MemRef<float, 2> output1(sizesOutput);
+  MemRef<float, 2> output2(sizesOutput);
+  MemRef<float, 2> input1(sizesOutput);
+  float *data = new float[outputRows * outputCols];
+  for (int i = 0; i < outputRows * outputCols; ++i) {
+    data[i] = 256;
+  }
+  MemRef<float, 2> copymemref(data, sizesOutput, 0);
+  for (int i = 0; i < outputRows * outputCols; ++i) {
+    data[i] = -1;
+  }
+  MemRef<float, 2> copymemref1(data, sizesOutput, 0);
+  if (option == BOUNDARY_OPTION::CONSTANT_PADDING) {
+    detail::_mlir_ciface_tophat_2d_constant_padding(
+        input, kernel, output, &output1, &output2, &input1, &copymemref,
+        &copymemref1, centerX, centerY, iterations, constantValue);
+  } else if (option == BOUNDARY_OPTION::REPLICATE_PADDING) {
+    detail::_mlir_ciface_tophat_2d_replicate_padding(
+        input, kernel, output, &output1, &output2, &input1, &copymemref,
+        &copymemref1, centerX, centerY, iterations, 0);
+  }
+}
+
+void BottomHat2D(Img<float, 2> *input, MemRef<float, 2> *kernel,
+                 MemRef<float, 2> *output, unsigned int centerX,
+                 unsigned int centerY, unsigned int iterations,
+                 BOUNDARY_OPTION option, float constantValue = 0) {
+  intptr_t outputRows = output->getSizes()[0];
+  intptr_t outputCols = output->getSizes()[1];
+  intptr_t sizesOutput[2] = {outputRows, outputCols};
+  MemRef<float, 2> output1(sizesOutput);
+  MemRef<float, 2> output2(sizesOutput);
+  MemRef<float, 2> input1(sizesOutput);
+  float *data = new float[outputRows * outputCols];
+  for (int i = 0; i < outputRows * outputCols; ++i) {
+    data[i] = -1;
+  }
+  MemRef<float, 2> copymemref(data, sizesOutput, 0);
```

Comment:
Try to use [this](https://github.com/buddy-compiler/buddy-mlir/blob/2ea25092dfca78cb4b18378e9c0017d71ed6ebca/lib/Interface/core/Container.cpp#L36) or similar constructors who let you construct a `MemRef` with default initial value instead of manually initialising things using loops.